### DISCUSSION TITLE
ESP32 internet radio based on VS1053

### DISCUSSION THREAD

**Discussion by nonix** [arendst/Tasmota#12126] [Ideas]
Hi guys, anyone tried (or is known to exists) a Internet radio based on VS1053 on ESP32 (I think it does work on ESP8266 as well but IMHO the CPU  is too weak to support both Tasmota and to feed the decoder)
Thx a lot,
 N.

**Comment by gemu2015** [comment]
the MAX98357 is already supported and provides mp3 and web radio support

**Comment by barbudor** [comment]
@nonix Did you meant components that are supported by Tasmota to build a web-radio ? 
Or an existing commercial product that can be tasmotized ?

@gemu2015 MAX98357 and VS1053 are too different beasts.
 MAX98357 is an DAC+Amp which requires the MP3/AAC/... to be decoded 1st (by software) and the PCM sample have to be sent through I2S to the MAX for audio rendering
 
VS1053 is a more versatile chip as it includes the MP3/AAC/WMA/FLAC/... decoder so it can be fed directly with the compressed stream and generate analog output as well as an I2S output. It can also play MIDI files (which are notes to be played, not even compressed audio).
VS1053 actually offload from the MCU the relatively heavy task of compressed audio decoding. 

Still, ESP32 can probably handle most of these codecs by software.
I would be more concerned that is quite different to standard Tasmota use-case and would probably require some tweaking due to the real-time contraints.

**Reply by sfromis** [reply]
Arguably, it has similarities with webcam real-time needs, including the processor doing a lot of work to keep up.

**Reply by nonix** [reply]
> 
> 
> @nonix Did you meant components that are supported by Tasmota to build a web-radio ?
> Or an existing commercial product that can be tasmotized ?
> 
> @gemu2015 MAX98357 and VS1053 are too different beasts.
> MAX98357 is an DAC+Amp which requires the MP3/AAC/... to be decoded 1st (by software) and the PCM sample have to be sent through I2S to the MAX for audio rendering
> 
> VS1053 is a more versatile chip as it includes the MP3/AAC/WMA/FLAC/... decoder so it can be fed directly with the compressed stream and generate analog output as well as an I2S output. It can also play MIDI files (which are notes to be played, not even compressed audio).
> VS1053 actually offload from the MCU the relatively heavy task of compressed audio decoding.
> 
> Still, ESP32 can probably handle most of these codecs by software.
> I would be more concerned that is quite different to standard Tasmota use-case and would probably require some tweaking due to the real-time contraints.

@barbudor I meant VS1053 Chip / module ( [VS1053 module on Alilexpress](https://www.aliexpress.com/item/32955766211.html?spm=a2g0o.productlist.0.0.83eef452Q2SYUn&algo_pvid=02e86dee-ad16-425c-b941-7a747b869b34&algo_expid=02e86dee-ad16-425c-b941-7a747b869b34-26&btsid=0bb0624516212704122665642e3ea3&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_) )
.. and yes I agree it deviates from standard tasmota use case. The idea is that I have most (all except one) IoTs run tasmota, this would be nice addition. And who knows, maybe one day it will be actual product ...

I will give it try to code a module, haven't yet though through the layout ...

Thank you for answering.
N.
